src/Extensions/Hosting.Services/HostBuilderExtensions.cs

```diff
@@ -25,57 +26,53 @@ public static class HostBuilderExtensions
 		/// <summary>
 		/// Add actions that will be executed inside service fabric service RunAsync method
 		/// </summary>
-		public static ServiceFabricHostBuilder<TContext> AddServiceAction<TContext>(
-			this ServiceFabricHostBuilder<TContext> builder,
-			Func<TContext, CancellationToken, Task> action)
-				where TContext : ServiceContext =>
```

Comment:
>where TContext : ServiceContext [](start = 4, length = 31)

You previously had a `where` clause in the code to narrow where the methods could be used. Is there any scope for doing something similar here again?

Reply:
Even narrowing to `class` could be beneficial, although further narrowing of course would be even better

---
In reply to: [395663670](https://github.com/microsoft/Omex/pull/172#discussion_r395663670) [](ancestors = 395663670)

Reply:
I've been trying to find something, but StatefulService and StatelessService don't have an interface or base class in common. There is an option of creating one but that will require to expose more classes and will complicate method declaration, because it will be `TService : IServiceFabricService<TContext>` so we would have to have one more generic parameter everywhere.
An argument in favor of leaving it like that is that this library controls the only possible values for this generics (it always used for sealed class with internal constructor), but it could be easily changed by somebody in the future.

Reply:
I agree with you that creating yet more classes for this purpose doesn't seem like an ideal solution. It's a pity there's nothing we can use, but it should be fine as is. Perhaps if you add the `<typeparam>` comment, you can indicate there the only supported values. That way you'll make it clear to callers that you won't support them if they pass in entirely wrong values. I realise it's internal for now, but as you said this could be easily changed, particularly if someone were to decide to say fork the repo.

Reply:
You are right, it makes sense here. I'll add them here. 
Probably I was afraid that if I'll add them in one case I would have to add it everywhere :)

Reply:
Added typecomments